sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/logical/expressions/Expression.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.logical.expressions;
```

Comment:
this should not be in the catalyst java package, which is meant to be private (although their visibility modifier doesn't always reflect that for a variety of reasons)

Reply:
Will do.

Reply:
+1, if it's going to be part of the table catalog API, I think `org.apache.spark.sql.catalog.v2.expressions` should be good

Reply:
Done.